test/cpp/end2end/spiffe_bundle_map_end2end_test.cc

```diff
@@ -0,0 +1,474 @@
+    }
+  }
+
+  TestServiceImpl service_;
+  std::unique_ptr<Server> server_ = nullptr;
+  std::thread* server_thread_ = nullptr;
+  std::string server_addr_;
+};
+
+void DoRpc(const std::string& server_addr,
+           const experimental::TlsChannelCredentialsOptions& tls_options,
+           bool expect_success, absl::string_view failure_message_regex = "",
+           StatusCode failure_code = StatusCode::OK) {
+  ChannelArguments channel_args;
+  channel_args.SetSslTargetNameOverride("foo.test.google.fr");
+  std::shared_ptr<Channel> channel = grpc::CreateCustomChannel(
+      server_addr, grpc::experimental::TlsCredentials(tls_options),
+      channel_args);
+
+  auto stub = grpc::testing::EchoTestService::NewStub(channel);
+  grpc::testing::EchoRequest request;
+  grpc::testing::EchoResponse response;
+  request.set_message(kMessage);
+  ClientContext context;
+  context.set_deadline(grpc_timeout_seconds_to_deadline(/*time_s=*/15));
+  grpc::Status result = stub->Echo(&context, request, &response);
+  if (expect_success) {
+    EXPECT_TRUE(result.ok()) << result.error_message().c_str() << ", "
+                             << result.error_details().c_str();
+    EXPECT_EQ(response.message(), kMessage);
+  } else {
+    EXPECT_FALSE(result.ok());
+    EXPECT_EQ(result.error_code(), failure_code);
+    EXPECT_THAT(result.error_message(),
+                ::testing::MatchesRegex(failure_message_regex));
+  }
+}
+
+TEST_F(SpiffeBundleMapTest, ServerSideSpiffeTLS) {
+  server_addr_ = absl::StrCat("localhost:",
+                              std::to_string(grpc_pick_unused_port_or_die()));
+  absl::Notification notification;
+  server_thread_ = new std::thread([&]() {
+    RunServer(&notification, kServerKeyPath, kServerCertPath, "",
+              kServerSpiffeBundleMapPath);
+  });
+  notification.WaitForNotification();
+
+  std::string root_cert =
+      grpc_core::testing::GetFileContents(std::string(kCaPemPath));
+  std::string client_key =
+      grpc_core::testing::GetFileContents(std::string(kClientKeyPath));
+  std::string client_cert =
+      grpc_core::testing::GetFileContents(std::string(kClientCertPath));
+  experimental::IdentityKeyCertPair key_cert_pair;
+  key_cert_pair.private_key = client_key;
+  key_cert_pair.certificate_chain = client_cert;
+  std::vector<experimental::IdentityKeyCertPair> identity_key_cert_pairs;
+  identity_key_cert_pairs.emplace_back(key_cert_pair);
+  auto certificate_provider =
+      std::make_shared<experimental::StaticDataCertificateProvider>(
+          root_cert, identity_key_cert_pairs);
+  grpc::experimental::TlsChannelCredentialsOptions options;
+  options.set_certificate_provider(certificate_provider);
+  options.watch_root_certs();
+  options.set_root_cert_name("root");
+  options.watch_identity_key_cert_pairs();
+  options.set_identity_cert_name("identity");
+
+  options.set_check_call_host(false);
+  auto verifier = std::make_shared<experimental::NoOpCertificateVerifier>();
```

Comment:
If we're setting a `NoOpCertificateVerifier`, why do we need to set the target override in `DoRpc`? I think we can likely omit one of the two, to make the test a bit simpler.